index.js

```diff
@@ -63,24 +43,20 @@ app.get('/product/:id', (req, res) => {
     productId = `P${userInput.padStart(3, '0')}`
   }
 
-  for (let product of productData.products) {
-    if (product.id === productId) {
-      jsonOutputObject = product;
-      break;
-    }
-  }
+  // Creating required Maps
+  dataService.getProducts;
+  dataService.getCategories;
+
+  jsonOutputObject = dataService.productMap[productId];
```

Comment:
Rename `jsonOutputObject` to something more meaningful.

Reply:
Done